pkg/git/git.go

```diff
@@ -598,3 +607,35 @@ func (r *Repo) tagsForBranch(branch string) ([]string, error) {
 
 	return strings.Fields(status.Output()), nil
 }
+
+// Add adds a file to the staging area of the repo
+func (r *Repo) Add(filename string) error {
```

Comment:
Do you think it would be useful to add tests for `Add` and `Commit`? Per usual, I would be fine with them coming in a follow-up PR.

Reply:
Yes sure, I added three more unit tests on top. I was not completely sure how to test a failing commit (should usually succeed), but I think this is not the usual case.